manifests/config.pp

```diff
@@ -1,5 +1,7 @@
-#
-class ntp::config inherits ntp {
+# Private Class
+class ntp::config (
+  $file_mode = $::ntp::params::config_file_mode,
```

Comment:
Oh, ntp::config isn't even a public class, so it shouldn't take any parameters. Just bring the variable straight into the namespace inside the main body.